app/uk/gov/hmrc/gform/eval/ExpressionResult.scala

```diff
@@ -123,6 +134,7 @@ sealed trait ExpressionResult extends Product with Serializable {
     case t: DateResult    => false
     case t: PeriodResult  => er.ifPeriodResult(t.value.toTotalMonths == _.toTotalMonths)
     case t: AddressResult => false
+    case t: ListResult    => false
```

Comment:
We should provide meaningful implementation for all operation.

This applies for all other 
```
case t: ListResult    => false
```

Reply:
Agreed, just added an implementation with tests